Description: Fix FTBFS due to erroneous nonnull annotation in glibc getcwd()
Author: Alex Murray <alex.murray@canonical.com>

This is still unresolved upstream but workaround it for now with this patch
to just disable the nonnull warnings via gcc pragmas around these call
sites

Upstream bug: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=96832
--- a/nss/coreconf/nsinstall/nsinstall.c
+++ b/nss/coreconf/nsinstall/nsinstall.c
@@ -236,14 +236,20 @@ main(int argc, char **argv)
 	return 0;
 
     if (!cwd) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnonnull"
 	cwd = GETCWD(0, PATH_MAX);
+#pragma GCC diagnostic pop
 	if (!cwd)
 	    fail("could not get CWD");
     }
 
     /* make sure we can get into todir. */
     xchdir(todir);
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnonnull"
     todir = GETCWD(0, PATH_MAX);
+#pragma GCC diagnostic pop
     if (!todir)
 	fail("could not get CWD in todir");
     tdlen = strlen(todir);

